### ISSUE TITLE
Output not visible while giving interactive inputs

### ISSUE THREAD

**Issue by thequantumquirk** [sindresorhus/execa#627]
When I try to execute any commands with interactive inputs like the next.js initialization command
It will not give me any outputs and just stay blank
But after I blindly select all the options it gives me all the outputs in the end as shown below:

![image](https://github.com/sindresorhus/execa/assets/67329471/1968140a-3d66-4ce9-859b-1c96a53b1724)

Is there any way to fix it?

**Comment by ehmicky** [comment]
Hi @thequantumquirk,

Thanks for reaching out.
This is most likely not a bug with Execa, but a general question on TTYs. 

For it to be considered a bug, would you be able to create a minimal reproduction example, with _no external dependencies_ except Execa? Thanks.

**Comment by thequantumquirk** [comment]
Yeah I'll do it asap.

**Comment by thequantumquirk** [comment]
> Hi @thequantumquirk,
> 
> Thanks for reaching out. This is most likely not a bug with Execa, but a general question on TTYs.
> 
> For it to be considered a bug, would you be able to create a minimal reproduction example, with _no external dependencies_ except Execa? Thanks.

Here's the link of the repo: https://github.com/thequantumquirk/execa-test/

```bash
npm install
node index.js
```

After running this the console output will be blank
when you press enter several times
the command will complete and show the output at the end including the interactive interactive inputs

**Comment by ehmicky** [comment]
A process' `stdout` is controlled by the `stdout` option.

It defaults to `pipe`, which means that `process.stdout` is set as a stream. With Execa, it also means the `stdout` result will be returned. It also means the process `stdout` is _not_ redirected to the parent process. In other words, you (intentionally) do not see the process's output in the console.

It can be set to `inherit` to redirect the child process' `stdout` to the parent's, i.e. to see the output in the console.

```js
await $({ stdout: 'inherit' })`npx create-next-app@latest my-app`;
```

Or:

```js
const $$ = $({ stdout: 'inherit' });
await $$`npx create-next-app@latest my-app`;
```

However, you won't be able to retrieve the result anymore. If you want to do both, you'll need to do this:

```js
const childProcess = $`npx create-next-app@latest my-app`;
childProcess.stdout.pipe(process.stdout);
const { stdout } = await childProcess
```

The behavior described above is not Execa-specific, this is how the `child_process` core module works. I.e. this is more a general support question about Node.js core behavior.

Please note that I am currently implementing https://github.com/sindresorhus/execa/issues/620, which will allow you, in the next release, to do this instead:

```js
const { stdout } = await $({ stdout: ['pipe', 'inherit'] })`npx create-next-app@latest my-app`;
```

Relevant documentation links: [`process.stdout`](https://github.com/sindresorhus/execa#stdout) and [`stdout` option](https://github.com/sindresorhus/execa#stdout-1).

**Comment by thequantumquirk** [comment]
> It can be set to `inherit` to redirect the child process' `stdout` to the parent's instead, i.e. to see the output in the console.
> 
> ```js
> await $({ stdout: 'inherit' })`npx create-next-app@latest my-app`;
> ```

This works like charm.
I think this should be featured in the README and/or may be a separate in-depth example.
Shall I create another issue so that you can assign it to me and I'll write the documentation for that?

**Comment by ehmicky** [comment]
Thanks @thequantumquirk!

However, we are currently working on significant changes of the `stdin`/`stdout`/`stderr` options, which will influence that documentation (or might even potentially remove its need), so we should wait on that first.

**Comment by ehmicky** [comment]
You might want to check out Execa 9.0.0: [release post](https://medium.com/@ehmicky/execa-9-release-d0d5daaa097f) and the [changelog](https://github.com/sindresorhus/execa/releases/tag/v9.0.0). In particular, the new `['pipe', 'inherit']` feature I mentioned above [is now available](https://github.com/sindresorhus/execa/blob/main/docs/output.md#multiple-targets). Also, the documentation for [inputs](https://github.com/sindresorhus/execa/blob/main/docs/input.md) and [outputs](https://github.com/sindresorhus/execa/blob/main/docs/output.md) has been improved, which should clarify the problems raised in this issue.
